Mathlib/CategoryTheory/Enriched/Limits/HasConicalTerminal.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2025 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson, Jon Eugster, Emily Riehl
+-/
+import Mathlib.CategoryTheory.Enriched.Limits.HasConicalProducts
+
+/-!
+## existence of conical terminal objects
+-/
+
+universe w v' v u u'
+
+namespace CategoryTheory.Enriched
+
+open Limits HasConicalLimit
+
+/-- A category has a conical terminal object
+if it has a conical limit over the empty diagram. -/
+abbrev HasConicalTerminal := HasConicalLimitsOfShape (Discrete.{0} PEmpty)
+
+variable (V : Type u') [Category.{v'} V] [MonoidalCategory V]
+variable (C : Type u) [Category.{v} C] [EnrichedOrdinaryCategory V C]
+
+example [HasConicalTerminal V C] : HasTerminal C := inferInstance
+
+/-! ### Conical Products -/
+
+example [HasConicalProducts.{0} V C] : HasConicalTerminal V C := inferInstance
```

Comment:
I've left this example in because I wanted to ask about this: after the instance below, it finds a different instance for this example and I think they might not be defeq (or are they?). Does this matter?

Reply:
`Has...Terminal` is a Prop, so that it cannot be an issue.